src/hooks/use-are-all-consents-accepted.ts

```diff
@@ -0,0 +1,28 @@
+import { useContext } from 'react'
+
+import { UsercentricsContext } from '../context.js'
+import { areAllConsentsAccepted } from '../utils.js'
+
+/**
+ * Whether all Usercentrics services have been given consent.
+ * Returns `true` or `false` based on consent status, or `null` when unknown (not yet loaded).
+ *
+ * @warn it's best to assume no consent until this hook returns `true`
+ */
+export const useAreAllConsentsAccepted = (): boolean | null => {
+    const { isClientSide, isInitialized } = useContext(UsercentricsContext)
+
+    /** Consent status is unknown during SSR because CMP is only available client-side */
+    if (!isClientSide) {
+        return null
+    }
+
+    /**
+     * Until Usercentrics CMP has loaded, return `null`
+     */
+    if (!isInitialized) {
+        return null
+    }
```

Comment:
Should we try to read from `localStorage` to find out the status before loading? [Similar implementation](https://github.com/s-group-dev/react-usercentrics/blob/e20843b910d4824f832f0e1ec158136831e0db5a/src/hooks/use-has-service-consent.ts#L25-L32) exists in `useHasServiceConsent()`. I guess in this  case it would look something like this:

```ts
    const { localStorageState } = useContext(UsercentricsContext)

    /**
     * Until Usercentrics CMP has loaded, try to get consent status from localStorage.
     * If it's not loaded, and there's nothing in localStorage, this will return `null`
     */
    if (!isInitialized) {
        return localStorageState.length > 0
            ? localStorageState.every((service) => service?.status === true)
            : null
    }
```

Reply:
The logic already exists in `useHasServiceConsent()` and was implemented to make it faster to start up logic based on it, so I would assume the same makes sense here. 👍

https://github.com/s-group-dev/react-usercentrics/blob/e20843b910d4824f832f0e1ec158136831e0db5a/src/hooks/use-has-service-consent.ts#L25-L32

and

https://github.com/s-group-dev/react-usercentrics/blob/e20843b910d4824f832f0e1ec158136831e0db5a/tests/hooks/use-has-service-consent.test.tsx#L62-L70